test/mountain_view/component_test.rb

```diff
@@ -16,16 +16,112 @@ def test_humanized_title
   def test_styleguide_stubs
     component = MountainView::Component.new("header")
 
-    expected_stub = {
-      id: 1,
-      title: "20 Mountains you didn't know they even existed",
-      subtitle: "Buzzfeed title"
-    }
+    expected_stub =
+      {
+        :meta => "There is this different classes",
+        :stubs =>
+          [
+            { :id  =>  1, :title => "20 Mountains you didn't know they even existed",
+             :subtitle => "Buzzfeed title"
+            },
+            { :id => 2, :title => "You won't believe what happened to this man at Aspen" }
+          ]
+      }
 
     assert_instance_of Array, component.styleguide_stubs
     assert_equal expected_stub, component.styleguide_stubs.first
   end
 
+  def test_component_stubs
+    component = MountainView::Component.new("header")
+    expected_stub =
+      [
+        { :id => 1, :title => "20 Mountains you didn't know they even existed",
+         :subtitle => "Buzzfeed title" },
+        { :id => 2, :title => "You won't believe what happened to this man at Aspen" }
+      ]
+
+    assert_instance_of Array, component.component_stubs
+    assert_equal expected_stub, component.component_stubs
+  end
+
+  def test_component_stubs?
+    component_with_stubs = MountainView::Component.new("header")
+    component_with_empty_stub_file = MountainView::Component.new("breadcrumbs")
+    component_without_stub_file =
+      MountainView::Component.new("social_media_icons")
+    compoenet_with_stubs_but_incorrect_format =
+      MountainView::Component.new('card')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.